### PR TITLE
Massive code cleanup

### PR DESCRIPTION
**Pull request by Brycey92**
I used IntelliJ IDEA's code analysis feature to help find warnings and bugs, and manually went through fixing them. The only warnings I let IDEA fix automatically were unused imports and most redundant casts.

ItemStacks have been converted to use ItemStack.EMPTY instead of null, to conform to new standards.

Annotations have been added to a LOT of the code to help devs determine when null can and can't be used, however they're nowhere near all-encompassing.

The inheritance of the BlockAdvancedBipropellantRocketMotor class was rearranged to prevent a potential mis-classification of them as monopropellant motors.

Vague fields and variables have been renamed where I found them, such as i -> itemStack or obj -> spaceObject. This excludes math variables that seemed intentionally named, and standardized names like loop control variables.

Lang files have been cleaned up to the best of my ability, but the non-en_US languages still have a lot of missing translation keys. Probably best for someone who speaks those languages to add the translations, rather than me with Google Translate.

There were some recipes that had invalid json, so I fixed them. This resulted in a few of the recipes changing slightly.

This commit should fix a lot of NPEs, though I didn't get them all.

**Next Steps:**
I highly recommend one of the original devs merge this and also run the source through their own IDE's code analysis, since there are still a lot of warnings I couldn't or didn't fix, and a large amount of them could be or turn into bugs.

The ItemStack changes affected some API methods. Mods developed against Advanced Rocketry should use builds containing this commit in the future, to prevent them from sending null ItemStacks. I did still split any changes to method names and prototypes into a separate branch that can be merged or ignored.

This commit requires a [sister-commit for libVulpes](https://github.com/Advanced-Rocketry/libVulpes/pull/50), since the ItemStack refactoring affected methods there too.

### CODE REVIEW

File: src/main/java/zmaster587/advancedRocketry/entity/EntityRocket.java [hunk 1]
```diff
@@ -1426,14 +1419,13 @@ private void reachSpaceUnmanned()
...
 				this.setPositionAndUpdate(pos.x, getEntryHeight(destinationDimId), pos.z);
-				return;
 			}
 			else {
 
 				//Make player confirm deorbit if a player is riding the rocket
 				if(hasHumanPassenger()) {
 					setInFlight(false);
-					pos.y = (float) getEntryHeight(destinationDimId);
+					this.setPositionAndUpdate(this.posX, getEntryHeight(destinationDimId), this.posZ);
```
Comment (Brycey92) [inline src/main/java/zmaster587/advancedRocketry/entity/EntityRocket.java:1428]: IDEA said this originally caused an NPE due to dereferencing `pos`. I hope my replacement doesn't change the functionality from what was intended

File: src/main/java/zmaster587/advancedRocketry/entity/EntityRocket.java [hunk 2]
```diff
@@ -1447,32 +1439,29 @@ private void reachSpaceUnmanned()
...
 				this.changeDimension(destinationDimId, pos.x, getEntryHeight(destinationDimId), pos.z);
-				return;
 			}
 			else {
 
 				//Make player confirm deorbit if a player is riding the rocket
 				if(hasHumanPassenger()) {
 					setInFlight(false);
-					pos.y = (float) getEntryHeight(destinationDimId);
+					this.setPositionAndUpdate(this.posX, getEntryHeight(destinationDimId), this.posZ);
```
Comment (Brycey92) [inline src/main/java/zmaster587/advancedRocketry/entity/EntityRocket.java:1458]: IDEA said this originally caused an NPE due to dereferencing `pos`. I hope my replacement doesn't change the functionality from what was intended

File: src/main/java/zmaster587/advancedRocketry/block/BlockAdvancedBipropellantRocketMotor.java [hunk 3]
```diff
@@ -4,7 +4,7 @@
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.World;
 
-public class BlockAdvancedBipropellantRocketMotor extends BlockRocketMotor {
+public class BlockAdvancedBipropellantRocketMotor extends BlockBipropellantRocketMotor {
```
Comment (Brycey92) [inline src/main/java/zmaster587/advancedRocketry/block/BlockAdvancedBipropellantRocketMotor.java:7]: This is the change to class inheritance that I mentioned in the commit and merge messages.

File: src/main/resources/assets/advancedrocketry/recipes/satellitebuilder.json [hunk 4]
```diff
@@ -18,36 +18,31 @@
...
+        "a":
         {
 	    "item": "minecraft:anvil",
 	    "data": 0
         },
-	"s":
-        {
-	    "item": "advancedrocketry:sawblade",
-	    "data": 0
-        },
```
Comment (Brycey92) [inline src/main/resources/assets/advancedrocketry/recipes/satellitebuilder.json:49]: This entry used the same key as the User Interface, meaning only this was part of the recipe. It seemed out of place, and this machine certainly seemed like it should have a User Interface in crafting, so I removed the saw blade. Let me know if this recipe should have included a saw blade.

**Comment by voidsong-dragonfly** [comment]
So I'm going to close this and ask you to run it against my NTR branch and _not_ against the main one at this moment, as that has a significant amount of changes.

**Comment by Brycey92** [comment]
Leaving my comments to the code here so I can easily copy them to the next PR.

File: src/main/resources/assets/advancedrocketry/recipes/satellitebuilder.json (hunk 4 above)
Comment (voidsong-dragonfly) [reply src/main/resources/assets/advancedrocketry/recipes/satellitebuilder.json:49]: For this I suggest checking what is was back on the master or 1.7.10 branch, as that would be before this move to JSON recipes could be made.

File: src/main/java/zmaster587/advancedRocketry/block/BlockAdvancedBipropellantRocketMotor.java (hunk 3 above)
Comment (voidsong-dragonfly) [reply src/main/java/zmaster587/advancedRocketry/block/BlockAdvancedBipropellantRocketMotor.java:7]: Don't think that's going to be a problem as long as it doesn't mess with the logic I wrote for it.

File: src/main/resources/assets/advancedrocketry/recipes/satellitebuilder.json (hunk 4 above)
Comment (Brycey92) [reply src/main/resources/assets/advancedrocketry/recipes/satellitebuilder.json:49]: Turned out to be the sawblade in 1.7.10, not the user interface. Changed in the new PR!

**Pull request closed without merging**